### DISCUSSION TITLE
Actions - Standalone and Resource

### DISCUSSION THREAD

**Discussion by ale1981** [laravel/nova-issues#3369] [Q&A] [answered]
Is it possible to have an action that is both standalone and can be used on a single or multiple resource?

I think its the same question as posted here https://github.com/laravel/nova-issues/issues/2211 and was marked as implemented but I can't see anything in the documentation on this?

**Comment by crynobone** [comment]
I believe you can register the action twice and handle both scenarios. 

Standalone actions will always receive an empty collection, and you can also check against `$this->isStandalone()` before trying to handle `$models`.

**Reply by ale1981** [reply]
Hi crynobone,

Thanks for your response. When I attempt to register the same action twice I get an "Sorry! You are not authorized to perform this action when running from the index with no resources selected?

    public function actions(Request $request)
    {
        return [
            Actions\SendPaymentReminder::make(),
            Actions\SendPaymentReminder::make()->standalone()
        ];
    }

**Reply by crynobone** [reply]
Just notice a bug based on your example, will submit a PR to fix this.

**Reply by ale1981** [reply]
Thanks, is it possible to link to the changes so I can apply them in my test env. ?

**Comment by crynobone** [comment, marked as answer]
This has been fixed with [v3.25.0 release](https://nova.laravel.com/releases/3.25.0).

Example:

```php
    /**
     * Get the actions available for the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function actions(Request $request)
    {
        return [
            new Actions\Sleep,
            Actions\Sleep::make()->standalone(),
        ];
    }
```

### Without selecting any resources

![image](https://user-images.githubusercontent.com/172966/116696565-a88f2580-a9f4-11eb-8ef5-c86937716b27.png)

### With selected resources

![image](https://user-images.githubusercontent.com/172966/116696534-a036ea80-a9f4-11eb-9453-c033b4b0172e.png)
